pkg/identity/basicallocator/basic_allocator.go

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package basicallocator
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/cilium/cilium/pkg/idpool"
+)
+
+// BasicIDAllocator is used for managing a range of identities (unsigned
+// integers) in a very simple way, by allocating specified or random identities,
+// and returning them to the available pool.
+// Concurrency control (mutex locks) is built into IDPool.
+type BasicIDAllocator struct {
+	idPool     *idpool.IDPool
+	minIDValue idpool.ID
+	maxIDValue idpool.ID
+}
+
+// NewBasicIDAllocator creates a new BasicIDAllocator for ID values between
+// the provided minIDValue and maxIDValue.
+func NewBasicIDAllocator(minIDValue, maxIDValue idpool.ID) *BasicIDAllocator {
+	return &BasicIDAllocator{
+		idPool:     idpool.NewIDPool(minIDValue, maxIDValue),
+		minIDValue: minIDValue,
+		maxIDValue: maxIDValue,
+	}
+}
+
+func (b *BasicIDAllocator) AllocateRandom() (idpool.ID, error) {
```

Comment:
Exported method here and below should have docs.

Reply:
Hey @derailed,
I've updated the docs.